src/control/security/README.md

```diff
@@ -1 +1,151 @@
+
+## Certificate Usage in DAOS
+
+Certificates are used in several locations to provide authentication and
+validation of user identity in a DAOS system.
+For information on how to set up the certificates in a DAOS system, see the
+[Admin Guide](/doc/admin/deployment.md#certificate-configuration).
+
+### Certificate Generation
+
+When generating certificates for a DAOS system, four types of certificates are
+generated.
+
+The first to be generated is the CA root certificate specific to this DAOS
+deployment. While it is advised to have per-system CA roots, there is no
+mechanism preventing reuse between DAOS deployments. The CA root is used to
+produce and validate the three remaining certificate types.
+
+The three remaining certificate types that are generated are the admin, agent,
+and server certificates. The admin certificate is used to protect the gRPC
+channel between the administrative node and also provide validation of the
+administrative user. The Agent certificate is used to ensure that the compute
+node belongs to the DAOS system and also provides validation of process identity
+credentials. The server certificate is used in three separate ways. First, it is
+used to protect the gRPC channel between the administrative node and the server,
+second, it protects the channel between DAOS servers, and finally, it provides
+signing functionality for data transfers between the server and compute nodes.
+
+DAOS provides a script and openssh configuration files for generating the
+various certificates needed by a DAOS installation. The script will craft a CA
+root certificate using the various configuration files found
+[in the DAOS source tree](/utils/certs).
+The config files contain the signing policies for the various other types of
+certificates as well, ensuring the cryptography used is as directed by current
+best practices for protecting TLS channels.
+
+For the remaining certificates, DAOS uses the common name to ensure that the
+correct certificates are used for the correct components. Initially, there will
+only be one class of administrative user, and the Common Name for the gRPC TLS
+channel will be admin, ensuring only admin certificates may be used to issue
+administrative commands to the control plane. This is to ensure that if a key
+and cert are obtained from a compute node that it cannot be used to connect to
+the administrative interface. Likewise, we ensure that the appropriate certs are
+used by the Agent and the Server by encoding their names into the Common Name.
+
+### Protecting Administrative Channels with Certificates
+
+Administration of a DAOS cluster will be performed by an administrator using the
+dmg utility and interfacing with the control plane service for the cluster. The
+control plane service is a distributed service provided by the DAOS cluster and
+resides on each node acting as a storage node. Even though the control plane
+service is present on every node, a subset of them is relegated as the
```

Comment:
s/relegated/designated/, I think?

Reply:
Fixed - you're right, wrong word there. :)